src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTaskLogic.cs

```diff
@@ -346,6 +347,27 @@ private void PopulateFrameworkAssemblyReferences(PackageBuilder builder, IPackTa
                             )));
         }
 
+        private void PopulateFrameworkReferences(PackageBuilder builder, IPackTaskRequest<IMSBuildItem> request)
+        {
+            var tfmSpecificRefs = new Dictionary<string, ISet<string>>(StringComparer.OrdinalIgnoreCase);
```

Comment:
Since you have so many places that specify the comparer and they should all be the same, perhaps you should make a constant for it (`FrameworkReferenceComparer`).